python/lsst/meas/extensions/shapeHSM/_hsm_moments.py

```diff
@@ -0,0 +1,417 @@
+
+
+class HsmMomentsPlugin(measBase.SingleFramePlugin):
+    """Base plugin for HSM adaptive moments measurement."""
+
+    ConfigClass = HsmMomentsConfig
+
+    def __init__(self, config, name, schema, metadata, logName=None):
+        super().__init__(config, name, schema, logName)
+
+        # Define flags for possible issues that might arise during measurement.
+        flagDefs = measBase.FlagDefinitionList()
+        self.FAILURE = flagDefs.addFailureFlag("General failure flag, set if anything went wrong")
+        self.NO_PIXELS = flagDefs.add("flag_no_pixels", "No pixels to measure")
+        self.NOT_CONTAINED = flagDefs.add(
+            "flag_not_contained", "Center not contained in footprint bounding box"
+        )
+        self.PARENT_SOURCE = flagDefs.add("flag_parent_source", "Parent source, ignored")
+        self.GALSIM = flagDefs.add("flag_galsim", "GalSim failure")
+        self.INVALID_PARAM = flagDefs.add("flag_invalid_param", "Invalid parameter")
+        self.EDGE = flagDefs.add("flag_edge", "Variance undefined outside image edge")
+        self.NO_PSF = flagDefs.add("flag_no_psf", "Exposure lacks PSF")
+
+        # Embed the flag definitions in the schema using a flag handler.
+        self.flagHandler = measBase.FlagHandler.addFields(schema, name, flagDefs)
+
+        # Utilize a safe centroid extractor that uses the detection footprint
+        # as a fallback if necessary.
+        self.centroidExtractor = measBase.SafeCentroidExtractor(schema, name)
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.SHAPE_ORDER
+
+    def _calculate(
+        self,
+        record: afwTable.SourceRecord,
+        *,
+        image: galsim.Image,
+        badpix: galsim.Image | None,
+        sigma: float,
+        precision: float = 1.0e-6,
+        centroid: Point2D,
+    ):
+        """
+        Calculate adaptive moments using GalSim's HSM and modify the record in
+        place.
+
+        Parameters
+        ----------
+        record : `~lsst.afw.table.SourceRecord`
+            Record to store measurements.
+        image : `~galsim.Image`
+            Image on which to perform measurements.
+        badpix : `~galsim.Image`, optional
+            Image representing bad pixels.
+        sigma : `float`
+            Estimate of object's Gaussian sigma.
+        precision : `float`, optional
+            Precision for HSM adaptive moments. Default is 1.0e-6.
+        centroid : `~lsst.geom.Point2D`
+            Centroid guess for HSM adaptive moments.
+
+        Raises
+        ------
+        MeasurementError
+            Raised for errors in measurement.
+        """
+        # Convert centroid to GalSim's PositionD type.
+        guessCentroid = galsim.PositionD(centroid.x, centroid.y)
+
+        try:
+            # Attempt to compute HSM moments.
+            shape = galsim.hsm.FindAdaptiveMom(
+                image,
+                weight=None,
```

Comment:
The weight needs to be supplied here. It's going to involve implementing the `convertMask` function in Python: https://github.com/lsst/meas_extensions_shapeHSM/blob/main/include/lsst/meas/extensions/shapeHSM/HsmAdapter.h#L82-L100

This would also mean that the method needs to take `weight` as an input argument.

Reply:
It seems like the GalSim version I'm looking at takes `weight` and `badpix` as `galsim.Image` objects, and it does the conversion on its own in `galsim.hsm.FindAdaptiveMom()`.